data_service/config/logging_config.py

```diff
@@ -1,69 +1,115 @@
-import json
+import re
 import sys
+import uuid
+import json
+import logging
+import datetime
+from time import perf_counter_ns
+from typing import Callable
 
-import json_logging
 import tomlkit
+from fastapi import Request
+from contextvars import ContextVar
 
 from data_service.config import environment
 
 
 def _get_project_meta():
-    with open("./pyproject.toml", encoding="utf-8") as pyproject:
+    with open("pyproject.toml", encoding="utf-8") as pyproject:
         file_contents = pyproject.read()
-
     return tomlkit.parse(file_contents)["tool"]["poetry"]
 
 
-pkg_meta = _get_project_meta()
-SERVICE_NAME = "data-service"
-host = environment.get("DOCKER_HOST_NAME")
-command = json.dumps(sys.argv)
-
+request_start_time: ContextVar[int] = ContextVar("request_start_time")
+correlation_id: ContextVar[str] = ContextVar("correlation_id")
+method: ContextVar[str] = ContextVar("method")
+url: ContextVar[str] = ContextVar("url")
+remote_host: ContextVar[str] = ContextVar("remote_host")
+response_status: ContextVar[int] = ContextVar("response_status")
+response_time_ms: ContextVar[int] = ContextVar("response_time_ms")
+
+
+class RequestInfoFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        # Make sure string only contains alphanumeric characters, underscores and/or dashes
+        record.correlation_id = re.sub(r"[^\w\-]", "", correlation_id.get(""))
```

Comment:
The correlation ID will usually be a UUID and [RFC](https://datatracker.ietf.org/doc/html/rfc4122.html#section-3) defines it as alphanumeric characters with dashes. It **can** be something else as this field can be set by the client, but will this be a problem?

Reply:
since it is set by the client I think it is best to do a quick sanitizing in case of a vulnerability in the python logger/printer/f-string

Reply:
Sounds good!